### PR TITLE
Render scripts to text

### PR DESCRIPTION
**Pull request by evmiguel**
This is the first iteration for rendering block scripts to test. The current implementation only renders the [first output example in the README](https://github.com/bocoup/sb-util#blockcollection). The current approach is to go down the block tree, and then follow each substack.

OUTPUT:
```
[
      "event_whenflagclicked",
      {
        "control_if_else": {
          "if": {
            "condition": {
              "sensing_keypressed": {
                "sensing_keyoptions": "space"
              }
            }
          },
          "then": [
            {
              "data_changevariableby": {
                "variable": "my variable",
                "value": "10"
              }
            },
            {
              "motion_gotoxy": {
                "X": "my variable",
                "Y": "10"
              }
            }
          ],
          "else": [
            {
              "control_wait_until": {
                "condition": {
                  "sensing_keypressed": {
                    "sensing_keyoptions": "space"
                  }
                }
              }
            }
          ]
        }
      }
    ]
```

### CODE REVIEW

File: src/sb-util.ts [hunk 1]
```diff
@@ -255,6 +257,150 @@ export class BlockCollection implements Queryable {
         const blocks: Iterable<BlockProperties> = storage.get(this);
         return map(blocks, (props: BlockProperties): Block => new Block(props));
     }
+
+    public renderToText(): Record<string, any> {
```
Comment (evmiguel) [inline src/sb-util.ts]: This is the entry point to renderToText()

File: src/sb-util.ts [hunk 2]
```diff
@@ -255,6 +257,153 @@ export class BlockCollection implements Queryable {
...
+            let substack2 = steps.pop();
+            if (typeof substack2 === 'string') {
+                substack2 = { [substack2]: {} };
+            }
+            const lastOpcode = Object.keys(substack2)[0];
+            Object.assign(substack2[lastOpcode], this.getSubstackBody(block, ControlBlocksArgs.SUBSTACK2));
+            steps.push(substack2);
+        }
+
+        const nextBlock = this.getNextBlock(block);
```
Comment (mzgoddard) [inline src/sb-util.ts]: Why do we include the next block? Shouldn't this just return the block we are pointing at and its substacks and reporters?

Comment (mzgoddard) [reply src/sb-util.ts]: If we want it to be the following blocks in the collection we should make it the blocks in the collection and not the next block. The collection may be a filtered set of blocks and so not directly reference each other.

Comment (mzgoddard) [reply src/sb-util.ts]: As is this should print the remaining contents of a single script in a scratch sprite. If the sprite has many scripts, it won't include those even if they are in the collection.

**Comment by evmiguel** [comment]
Closing this until we have AST generatorion

**Pull request closed without merging**